processing/data_collection/gazette/spiders/mg_governador_valadares.py

```diff
@@ -0,0 +1,113 @@
+import ast
+from datetime import datetime, date
+import json
+import re
+
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class MgGovernadorValadares(BaseGazetteSpider):
+    TERRITORY_ID = "3127701"
+    name = "mg_governador_valadares"
+    allowed_domains = ["valadares.mg.gov.br"]
+    start_urls = [
+        "https://www.valadares.mg.gov.br/diario-eletronico/caderno/diario-oficial-eletronico/1"
+    ]
+
+    ITEMS_PER_PAGE = "100"
+    BASE_URL = "https://www.valadares.mg.gov.br/"
+
+    path = ""
+    current_page = 0
+
+    def parse(self, response):
+        scripts = response.xpath("//script//@src").extract()
+        endpoint = next(path for path in scripts if "diel_diel_lis" in path)
+        self.path = endpoint
+        yield self.make_request(self.current_page)
+
+    def parse_items(self, response):
+        body = response.body
+        has_no_results = body == "null;/*".encode()
+        if has_no_results:
+            return
+
+        content = re.findall(
+            "new Ajax\.Web\.DataTable\((?P<conteudo>.*)\);", body.decode("utf-8")
+        )[0]
+        content = content.replace("new Date", "")
+
+        rows = None
+        definition = None
+
+        try:
+            definition, rows = self.extract_definitions_and_rows(content)
+        except Exception as e:
+            return
```

Comment:
I would move this code to another function with a meaningful name.

Reply:
This is part of `parse_definitions_and_rows` now.